app/routes.py

```diff
@@ -1,2 +1,256 @@
-from flask import Blueprint
+from flask import Blueprint, request, jsonify, make_response
+from werkzeug.datastructures import Authorization
+from app import db
+from app.models.customer import Customer
+from app.models.video import Video
+from app.models.rental import Rental
+from datetime import datetime, timedelta
 
+customers_bp = Blueprint("customers", __name__, url_prefix="/customers")
+videos_bp = Blueprint("videos", __name__, url_prefix="/videos")
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+@customers_bp.route("/<customer_id>", methods = ["GET", "PUT", "DELETE"])
+def get_single_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    request_body = request.get_json()
+    if customer is None:
+        return jsonify({"details": "Customer does not exist"}), 404
+    elif request.method == "GET":
+        return jsonify(customer.to_json()), 200
+    elif customer.name is None or customer.name is not str(customer.name):
+        return jsonify({"details": "Name data is missing"}), 400
+    elif customer.phone is None:
+        return jsonify({"details": "Phone data is missing"}), 400
+    elif customer.postal_code is None:
+        return jsonify({"details": "Postal_code data is missing"}), 400
+    elif request.method == "PUT":
+        if all(keys in request_body for keys in ("name","postal_code","phone")) == True:
+            customer.name = request_body["name"]
+            customer.postal_code = request_body["postal_code"]
+            customer.phone = request_body["phone"]
+
+            db.session.add(customer)
+            db.session.commit()
+            return jsonify(customer.to_json()), 200
+        else:
+            return jsonify({"error": "Bad Request"}), 400
+    elif request.method == "DELETE":
+        db.session.delete(customer)
+        db.session.commit()
+        return { "id" : customer.customer_id }, 200
+
+@customers_bp.route("", methods=["GET"])
+def customers_index():
+    customers = Customer.query.all()
+    if customers == None:
+        return [], 200
+    else:
+        customers_response = []
+        for customer in customers:
+            customers_response.append(customer.to_json())
+        return jsonify(customers_response), 200
```

Comment:
one nice thing about `query.all()` is that if no data is returned, it will still create an empty list for us, so the `if` statement won't ever be triggered. Maybe we could do something like this:

```suggestion
    customers = Customer.query.all()
    if len(customers) < 1:
        return customers, 200
    else:
        customers_response = []
        for customer in customers:
            customers_response.append(customer.to_json())
        return jsonify(customers_response), 200
```